src/scenario/scenario_events_parameter_data.c

```diff
@@ -806,3 +806,363 @@ void scenario_events_parameter_data_get_display_string_for_value(parameter_type
             }
     }
 }
+
+static uint8_t *append_const_text(const uint8_t *text_to_append, uint8_t *result_text, int *maxlength)
+{
+    int text_length = string_length(text_to_append);
+    result_text = string_copy(text_to_append, result_text, *maxlength);
+    *maxlength -= text_length;
+    return result_text;
+}
+
+static uint8_t *append_text(uint8_t *text_to_append, uint8_t *result_text, int *maxlength)
+{
+    int text_length = string_length(text_to_append);
+    result_text = string_copy(text_to_append, result_text, *maxlength);
+    *maxlength -= text_length;
+    return result_text;
+}
```

Comment:
Why the different functions? The first one would work fine as a non-`const` is automatically cast to `const`.

Just use `static uint8_t *append_text(const uint8_t *text_to_append, uint8_t *result_text, int *maxlength)`

Reply:
I did not know that. Thanks. Updated.